doc/architecture/decisions/0002_custom_forms.md

```diff
@@ -0,0 +1,63 @@
+# Custom forms
+
+Date: 2023-09-05
+
+## Status
+
+Accepted
+
+## Context
+
+We are using the ["Task List System"](doc/task_list_system.md) to generate most
+of our form pages. This system allows us to create 'pages' that use a shared
+form an pattern of paths to PUT their data to:
+
+**1. Regular 'page' submission to the pages controller**
+
+```ts
+paths.applications.pages.update({
+  id: applicationId, 
+  task: task, 
+  page: pageName
+})
+```
+
+e.g.
+
+`PUT /applications/{applicationId}/tasks/confirm-eligibility/pages/confirm-eligibility`
+
+In this case the task and page information is in the path.
+
+**2. Special cases of submission direct to the applications controller**
+
+We have come across a few instances where we need to transform or populate data
+in a different way than the regular form pages. E.g. The `oasys-import` page in
+Risk to Self needs to add data under the keys of other pages. In these cases we
+are not using the shared `form` component from
+`server/views/applications/pages/layout.njk` but instead rolling a new form that
+posts directly to a path on the applications controller. 
+
+```ts
+paths.applications.update({ 
+  id: applicationId 
+})
+```
+
+e.g. 
+
+`PUT /applications/{applicationId}`
+
+In this case the task and page data is supplied in hidden form fields.
+
+## Decision
+
+We will submit form data direct to the application controller in more complex
+scenarios e.g.
+
+- when we are importing data in bulk, behind the scenes, from upstream services like OASys
+- when we need to build up an arbitrarily long list of items, as in this case
+  where the applicant can add one ACCT record after the other.
+
+Where relevant we will put these forms in their own folder, e.g. a `custom-data`
```

Comment:
I like this idea, it's very clear signposting, but could we put these in a `custom-data` data directory within the task directory they belong in? Then we can preserve both categories (task, and form type) in organising these files.

Reply:
If I understand correctly, we suggest `custom-data` as the directory name in the ADR but use `custom-forms` in the commit?

Reply:
> If I understand correctly, we suggest custom-data as the directory name in the ADR but use custom-forms in the commit?

whoops! changed to `custom-forms`
I also want to rename the `acct-data` page to something more like `add-acct` but figured this PR was big enough already and can do it in a subsequent one :)

Reply:
Ah I think I misread the file directory system and you were already doing what I suggested 🙂